lib/useOnyx.ts

```diff
@@ -33,6 +32,8 @@ type UseOnyxInitialValueOption<TInitialValue> = {
     initialValue?: TInitialValue;
 };
 
+type UseOnyxSelector<TKey extends OnyxKey, TReturnValue = OnyxValue<TKey>> = (data: OnyxValue<TKey> | undefined) => TReturnValue;
```

Comment:
```suggestion
type UseOnyxSelector<TKey extends OnyxKey, TReturnValue> = (data: OnyxValue<TKey>) => TReturnValue;
```

We don't need to specify `undefined` as `OnyxValue` already includes it. Also, no need to specify default for `TReturnValue`

Reply:
Got this error when trying to remove `undefined` from union:

```
Argument of type 'OnyxValue<TKey> | undefined' is not assignable to parameter of type 'OnyxValue<TKey>'.
Type 'undefined' is not assignable to type 'OnyxValue<TKey>'.ts(2345)
  ```

Reply:
I left some new comments that will solve this!